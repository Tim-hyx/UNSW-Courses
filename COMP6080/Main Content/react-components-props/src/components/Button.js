import React from 'react';

const Button = ({ background, click, title }) => {
  return (
  	<button
  	  style={{
  	    width: '100px',
  	    background: background
  	  }}
  	  onClick={click}
  	>
      {title ? title : 'Submit'}
    </button>
  );
}

export default Button; 