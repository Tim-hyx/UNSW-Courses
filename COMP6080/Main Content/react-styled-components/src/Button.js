import styled, { css, keyframes } from 'styled-components';

const removeDefaultButtonStyles = css`
  border: 0;
  padding: 0;
  font-family: inherit;
  font-weight: normal;
  background-color: unset;
  outline: none;
`;

export const BaseButton = styled.button`
  ${removeDefaultButtonStyles};
  cursor: pointer;
  padding: 12px;
  box-sizing: border-box;
  border-radius: 4px;
  font-weight: bold;
`;

const clickAnimation = keyframes`
  0% {
    transform: scale(1);
    border-radius: 4px;
    
  }

  50% {
    transform: scale(1.01);
  }

  75% {
    transform: scale(0.99);
  }

  100% {
    transform: scale(1);
  }
`;

function createHoverableButton(backgroundColor, hoverColor, inverted) {
  return styled(BaseButton)`
    color: ${inverted ? 'white' : 'black'};
    background-color: ${backgroundColor};

    &:hover {
      background-color: ${hoverColor};
      animation: ${props => props.animated ? css`${clickAnimation} 0.15s ease-in-out` : undefined};
    }
  `;
}

export const PrimaryButton = createHoverableButton(
  "#19aaee",
  "#33bbff",
  true,
);

export const SecondaryButton = createHoverableButton(
  "#d5d7d7",
  "#edf0f2",
  false,
);

export const ContrastButton = createHoverableButton(
  "#7d2ae8",
  "#8d39fa",
  true,
);
