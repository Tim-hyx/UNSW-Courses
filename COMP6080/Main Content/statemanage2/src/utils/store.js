import React from 'react'

export const StoreContext = React.createContext(null)

export default ({ children }) => {
  const [movies, setMovies] = React.useState([])
  const [tvShows, setTvShows] = React.useState([])

  const store = {
    movies: [movies, setMovies],
    tvShows: [tvShows, setTvShows],
  }

  return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>
}