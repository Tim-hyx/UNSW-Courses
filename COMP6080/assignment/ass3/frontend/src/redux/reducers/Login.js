import { userConstants } from "../actionTypes";

// check the localStorage see if user already logged in
const accessToken = localStorage.getItem("accessToken");

// Shorthand property names stynex
const initialState = accessToken
    ? { loggedIn: true, accessToken }
    : { loggedIn: false };

// given the current state and action return the next state
export default (state = initialState, action) => {
    switch (action.type) {
    case userConstants.LOGIN_REQUEST:
        return {
            loggingIn: true,
        };
    case userConstants.LOGIN_SUCCESS:
        return {
            loggedIn: true,
            accessToken: action.accessToken,
        };
    case userConstants.LOGIN_FAILURE:
        return {};
    case userConstants.LOGOUT_REQUEST:
        return { ...state, loggingOut: true };
    case userConstants.LOGOUT_SUCCESS:
        return { loggedIn: false };
    case userConstants.LOGOUT_FAILURE:
        return state;
    default:
        return state;
    }
};
