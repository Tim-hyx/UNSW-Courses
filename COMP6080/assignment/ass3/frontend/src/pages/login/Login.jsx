import React, {useState} from "react";
import {makeStyles} from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import InputAdornment from "@material-ui/core/InputAdornment";
import EmailIcon from "@material-ui/icons/Email";
import LockIcon from "@material-ui/icons/Lock";
import {useDispatch} from "react-redux";
import {useHistory} from "react-router-dom";
import {login} from "../../redux/actions";

const useStyles = makeStyles(() => ({
    root: {
        // 64px is the height of navbar
        height: "calc(100vh - 64px)",
        justifyContent: "center",
        alignItems: "center",
    },
    form: {
        display: "flex",
        flexDirection: "column",
        maxWidth: 400,
        minWidth: 300,
    },
}));
const Login = () => {
    const [email, setEmail] = useState();
    const [password, setPassword] = useState();
    const classes = useStyles();
    const dispatch = useDispatch();
    const history = useHistory();
    return (
        <Grid container className={classes.root}>
            <form
                className={classes.form}
                onSubmit={(e) => {
                    e.preventDefault();
                    dispatch(login(email, password));
                    history.push("/dashboard");
                }}
            >
                <TextField
                    label="Email"
                    type="email"
                    margin="normal"
                    required
                    InputProps={{
                        startAdornment: (
                            <InputAdornment position="start">
                                <EmailIcon/>
                            </InputAdornment>
                        ),
                    }}
                    onChange={(event) => setEmail(event.target.value)}
                />
                <TextField
                    label="Password"
                    margin="normal"
                    type="password"
                    required
                    InputProps={{
                        startAdornment: (
                            <InputAdornment position="start">
                                <LockIcon/>
                            </InputAdornment>
                        ),
                    }}
                    onChange={(event) => setPassword(event.target.value)}
                />
                <div style={{height: 20}}/>
                <Button color="primary" variant="contained" type="submit">
                    Log in
                </Button>
            </form>
        </Grid>
    );
};

export default Login;
