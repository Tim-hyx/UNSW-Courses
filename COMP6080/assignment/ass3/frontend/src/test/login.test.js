import {mount} from 'enzyme';
import React from "react";
import {Provider} from "react-redux";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import { MemoryRouter as Router } from 'react-router-dom';
import sinon from 'sinon';
import store from "../redux/stores";
import Login from "../pages/login/Login";


// test login page
describe('test login page static html', () => {


    let loginPage;
    let wrapper;
    beforeEach(()=>{
        wrapper= mount(
            <Provider store={store}>
                <Login  />
            </Provider>);
        loginPage = wrapper.children();
    });
    afterEach(() => {
        wrapper.unmount();
	 });

    
    it('render a button to login', () => {
       
       
        const loginButton = loginPage.find(Button);
        expect(loginButton.text()).toBe("Log in");
        expect(loginButton).toMatchSnapshot();
    })

    it('render a textfield to input user email', () => {
      
        expect(loginPage.find(TextField).first().text()).toBe("Email *");
    });

    it('render a textfield to input user password ', () => {
      
        expect(loginPage.find(TextField).last().text()).toBe("Password *");
    });
    
    


})


describe('test login page dynamic behavior', () => {
    it('do not submit the form when click the submit button', () => {
	
        const onSubmit = sinon.spy();
        const mountWithRouter = node => mount(
            <Router>
                <Provider store={store}>
                    {node}
                </Provider>
            </Router>
        );
        const wrapper2 = mountWithRouter(<Login onSubmit={onSubmit} />);
        const form = wrapper2.find('form');
        form.simulate('submit');

        // because we build a SPA in react
        // when the form submit, we use preventDefault
        // and use react router to direct page without refresh
        expect(onSubmit.called).toBe(false);
    })
})