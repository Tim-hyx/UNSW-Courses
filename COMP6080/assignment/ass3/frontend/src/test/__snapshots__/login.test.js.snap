// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test login page static html render a button to login 1`] = `ReactWrapper {}`;
